bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -243,6 +243,7 @@ UCS_PROFILE_FUNC(jobject, process_request, (request, callback), void *request, j
         } else {
             // request was completed whether by progress in other thread or inside
             // ucp_tag_recv_nb function call.
+            set_jucx_request_completed(env, jucx_request, ctx);
```

Comment:
shouldn't the progress thread set it to completed?

Reply:
No in case it came before process request, it'll just set status: https://github.com/openucx/ucx/blob/master/bindings/java/src/main/native/jucx_common_def.cc#L204